### ISSUE TITLE
Improve error reporting when display configuration won't do

### ISSUE THREAD

**Issue by Saviq** [MirServer/mir#2600]
We had a case of hardware that couldn't drive the default display config Mir came up with (based on the default / best modes for the displays).

The errors raised was somewhat unhelpful:
```
Mir fatal error: Output HDMI-A-1 has no associated CRTC to wait on
# or
< - ERROR - > gbm-kms: Output HDMI-A-1 has no associated CRTC to schedule page flips on
```

In this case it was about bandwidth, ideally we could try other modes (bisect the available ones?).

I'm told Atomic KMS could also improve this.

**Comment by AlanGriffiths** [comment]
Similar to #2563
